src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepRestartTest.java

```diff
@@ -64,6 +78,81 @@ public void restartBetweenJobs() throws Throwable {
+            upstream.addProperty(new DurabilityHintJobProperty(FlowDurabilityHint.PERFORMANCE_OPTIMIZED)); // TODO: Unnecessary once we pick up https://github.com/jenkinsci/workflow-cps-plugin/pull/807.
+            upstream.setDefinition(new CpsFlowDefinition("build(job: 'downstream', wait: false)", true));
+            WorkflowRun upstreamRun = j.buildAndAssertSuccess(upstream);
+            // Check action while the build is still in the queue.
+            String buildStepId = BuildTriggerStepTest.findFirstNodeWithDescriptor(upstreamRun.getExecution(), BuildTriggerStep.DescriptorImpl.class).getId();
+            DownstreamBuildAction action = upstreamRun.getAction(DownstreamBuildAction.class);
+            assertThat(action.getFlowNodeId(), equalTo(buildStepId));
+            assertThat(action.getJobFullName(), equalTo(downstream.getFullName()));
+            assertThat(action.getBuildNumber(), nullValue());
+            assertThat(action.getBuild(), nullValue());
+            // Check again once the build has started.
+            j.createOnlineSlave(Label.parseExpression("agent"));
+            await().atMost(10, TimeUnit.SECONDS).until(() -> action.getBuildNumber(), notNullValue());
+            Run<?, ?> downstreamRun = downstream.getLastBuild();
+            assertThat(action.getFlowNodeId(), equalTo(buildStepId));
+            assertThat(action.getJobFullName(), equalTo(downstream.getFullName()));
+            assertThat(action.getBuildNumber(), equalTo(downstreamRun.getNumber()));
+            assertThat(action.getBuild(), equalTo(downstreamRun));
+            j.waitForCompletion(downstreamRun);
+        });
+        sessions.then(j -> {
+            FreeStyleProject downstream = j.jenkins.getItemByFullName("downstream", FreeStyleProject.class);
+            WorkflowJob upstream = j.jenkins.getItemByFullName("upstream", WorkflowJob.class);
+            WorkflowRun upstreamRun = upstream.getLastBuild();
+            String buildStepId = BuildTriggerStepTest.findFirstNodeWithDescriptor(upstreamRun.getExecution(), BuildTriggerStep.DescriptorImpl.class).getId();
+            DownstreamBuildAction action = upstreamRun.getAction(DownstreamBuildAction.class);
+            assertThat(action.getFlowNodeId(), equalTo(buildStepId));
+            assertThat(action.getJobFullName(), equalTo(downstream.getFullName()));
+            assertThat(action.getBuildNumber(), equalTo(downstream.getLastBuild().getNumber()));
+            assertThat(action.getBuild(), equalTo(downstream.getLastBuild()));
+        });
+    }
+
+    @Test
+    public void downstreamBuildActionMultipleBuilds() throws Throwable {
+        sessions.then(j -> {
+            FreeStyleProject downstream = j.createFreeStyleProject("downstream");
+            WorkflowJob upstream = j.jenkins.createProject(WorkflowJob.class, "upstream");
+            upstream.setDefinition(new CpsFlowDefinition("build(job: 'downstream'); build(job: 'downstream')", true));
```

Comment:
or more idiomatically
```suggestion
            upstream.setDefinition(new CpsFlowDefinition("build 'downstream'; build 'downstream'", true));
```